src/05-derivatives/05-diffusion-derivatives.md

```diff
@@ -63,141 +63,128 @@ field. If a new model is used, common sense should be used to derive a name
-| ------------ | ------------------------------------------------------------------------------------------------------------------------------------ |
-| Model        | Model label - see table above                                                                                                        |
-| Shells       | OPTIONAL. Shells that were utilized to fit the model, as a list of b-values. If the key is not present, all shells were used.        |
-| Gradients    | OPTIONAL. Subset of gradients utilized to fit the model, as a list of three-elements lists. If not present, all gradients were used. | ModelDescription | OPTIONAL. Extended information to describe the model. |
-| ModelURL     | OPTIONAL. URL to the implementation of the specific model utilized.                                                                  |
-| Parameters   | OPTIONAL. A dictionary of model parameters (see below for keys specific to each model)                                               |
+| **Key name**     | **Description**                                                                                                                      |
+| ---------------- | ------------------------------------------------------------------------------------------------------------------------------------ |
+| Shells           | OPTIONAL. Shells that were utilized to fit the model, as a list of b-values. If the key is not present, all shells were used.        |
+| Gradients        | OPTIONAL. Subset of gradients utilized to fit the model, as a list of three-elements lists. If not present, all gradients were used. |
+| ModelDescription | OPTIONAL. Extended information to describe the model.                                                                                |
+| ModelURL         | OPTIONAL. URL to the implementation of the specific model utilized.                                                                  |
+| Parameters       | OPTIONAL. A dictionary of model parameters (see below).                                                                              |
 
--   Model:
+Parameters that may be stored within the JSON sidecar "Parameters" field depend on the particular model used:
 
-    -   `DTI` :
+-   `DTI` :
 
-        -   `FitMethod` : {`WLS`,`OLS`,`NLLS`,`RESTORE`}
-        -   `RESTORE` : sigma
+    -   `FitMethod` : {`WLS`,`OLS`,`NLLS`,`RESTORE`}
+    -   `RESTORE` : sigma
 
-    -   `DKI` and `WMTI`:
+-   `DKI` and `WMTI`:
 
-        -   `FitMethod` : {`WLS`,`OLS`,`NLLS`}
+    -   `FitMethod` : {`WLS`,`OLS`,`NLLS`}
 
-    -   `CSD`:
+-   `CSD`:
 
-        -   `SphericalHarmonicOrder` : value
-        -   `ResponseFunctionOrder` : value
-        -   `ResponseFunction` : \[1:response_function_order_value,b0mean\]
-        -   `Basis` : {`MRtrix 0.2`,`MRtrix3`,`DESCOTEAUX`}
+    -   `Tissue` : string
+    -   `SphericalHarmonicDegree` : value
+    -   `ResponseFunctionDegree` : value
+    -   `ResponseFunction` : values

```

Comment:
I wonder whether this occasion could be used to bring consensus on these formats and increase compatibility between MRtrix and Dipy. 
@arokem @Lestropie

Reply:
Sorry - I am not sure that I follow. What exactly do you propose? That we change the implementation in DIPY? It is just a simple matter of programming.

Reply:
I personnally don't think the model should be changed in dipy. I think we should support both types in the spec, since they are quite different.

Reply:
I was not familiar with the Dipy representation, so treat my comments as being naive of such.

While eigenvalues of a tensor can be converted to a (zonal) spherical harmonic representation, the converse is not true; so the latter is technically more "general". It's also the case that in *MRtrix3* one does not actually *need* a *b*=0 "shell": the rows simply correspond to ascending *b*-value "shells" (where we treat *b*=0 as a "shell"). So e.g. something we've considered in the past is acquiring 3-6 volumes at *b*~100-200 to get comparable contrast but with less Gibbs ringing.

So from this, it would certainly be problematic to *only* have the Dipy representation in the BIDS spec, as it would fundamentally preclude *MRtrix3* data from being properly represented. Having the two side-by-side in the spec would certainly be feasible; the question then is whether that complexity is warranted, or whether it would in fact be "cleaner" for Dipy to support the latter representation (even if it were to just perform conversions to its own internal representation at import/export).

Having said that: If this is for the purpose of deconvolution, is the tensor representation not immediately converted to (zonal) spherical harmonics in the Dipy code anyway? If that's the case, it might be overall more convenient to move to that representation? The *b*=0 mean is still encoded as the l=0 term of the first row if there is at least one *b*=0 volume in the input data.

-----

One other thing I've realised here is that the tissue response function cannot be properly interpreted unless the *b*-value represented by each row is known. While this information is *kind of* in the `bvals` file, there can be ambiguity in regards to how these data are arranged into "shells". So it might be preferable to explicitly encode such here.

Reply:
@arokem Do you know from memory how this is done in Dipy? Else I'll recheck.

Reply:
Two things I now notice about the DIPY implementation. The first is that we support two kinds of inputs for response functions (at least in the programmatic API, probably not in the CLI). The first is a spherical harmonic representation through an [`AxSymShResponse` object](https://github.com/nipy/dipy/blob/master/dipy/reconst/csdeconv.py#L28-L59). The other is a tensor (three eigenvalues). They both require an S0 value. 

But yes, IIUC, the tensor representation is almost immediately converted into a SH representation ([here](https://github.com/nipy/dipy/blob/master/dipy/reconst/csdeconv.py#L160-L168)). Is that representation identical to the one that mrtrix has?

Reply:
There's a couple of different ways that this could be done if both response function representations were to be supported:

-----

`ResponseFunctionZSH` : values
        (1 row per unique *b*-value, with lower *b*-values listed first;
        1 column per even harmonic degree starting from zero)
`ResponseFunctionTensor` : values
        (4 values: 3 tensor eigenvalues, then a b=0 intensity reference)

Obviously these two fields would need to be mutually exclusive.

-----

`ResponseFunctionBasis`: [`ZSH`,`Tensor`]
`ResponseFunctionCoeffs` : values
        (If "`ResponseFunctionBasis`" = "`ZSH`":
        1 row per unique *b*-value, with lower *b*-values listed first,
        1 column per even harmonic degree starting from zero;
        If "`ResponseFunctionBasis`" = "`Tensor`":
        4 values: 3 tensor eigenvalues, then a b=0 intensity reference)

-----

Additionally, would field "`ResponseFunctionDegree`" need to be removed? In the case of the zonal spherical harmonic representation, the maximal degree can be inferred from the coefficients; in the case of the tensor representation, it either doesn't apply to those coefficients directly, or if that parameter is supposed to apply then to the initial conversion of those coefficients to the ZSH basis, it re-raises the question of why the response function would be provided in the tensor form in the first place.

Reply:
I've given some of this a go in 32183c2; see what you think.